### PR TITLE
Wip filelock deadlock

### PR DESCRIPTION
**Pull request by ukernel**

### CODE REVIEW

**Comment by ukernel** [comment]
http://pulpito.ceph.com/zyan-2016-03-02_06:21:11-fs-wip-zyan-testing-testing-basic-mira/

**Comment by gregsfortytwo** [comment]
@jcsp, please review

File: src/mds/flock.h [hunk 1]
```diff
@@ -29,17 +29,52 @@ inline bool ceph_filelock_owner_equal(ceph_filelock& l, ceph_filelock& r)
...
-    l.length == r.length &&
-    l.type == r.type &&
-    ceph_filelock_owner_equal(l, r);
+inline int ceph_filelock_owner_compare(const ceph_filelock& l, const ceph_filelock& r)
+{
+  if (l.client != r.client)
+    return l.client > r.client ? 1 : -1;
+  if (l.owner != r.owner)
+    return l.owner > r.owner ? 1 : -1;
+  if (l.owner & (1ULL << 63))
```
Comment (jcsp) [inline src/mds/flock.h]: It would be good to have a descriptive-named #define for this 1<<63 value

**Comment by jcsp** [comment]
This looks sane to me, I'm inclined to merge once tests are run, I've included this on just-pushed-to-gitbuilders wip-jcsp-testing branch

File: src/mds/flock.h (hunk 1 above)
Comment (gregsfortytwo) [inline src/mds/flock.h]: Or just copy the comment from ceph_filelock_owner_equal up above; it confused the heck out of me until I looked at the full file and saw that. :)

**Comment by jcsp** [comment]
Reviewed-by

**Comment by gregsfortytwo** [comment]
sounds good, it's passed through my branch as well

**Pull request merged**